### ISSUE TITLE
Include Terraform version in module source http get request

### ISSUE THREAD

**Issue by antonbabenko** [hashicorp/terraform#14753]
When using module source with http it would be very helpful to also receive Terraform version as an additional GET parameter.

Current access log:
```
127.0.0.1 - - [19/May/2017:22:02:15 +0200] "GET /tf-module/v1.0.0.html?terraform-get=1 HTTP/1.1" 200 453 "-" "Go-http-client/1.1"
```

Desired access log:
```
127.0.0.1 - - [19/May/2017:22:02:15 +0200] "GET /tf-module/v1.0.0.html?terraform-get=1&terraform-version=v0.9.5 HTTP/1.1" 200 453 "-" "Go-http-client/1.1"
```

**Comment by apparentlymart** [comment]
Hi @antonbabenko! Thanks for this request.

I'm feeling hesitant to change the established URL scheme at this point, in case it breaks people doing simplistic matching/routing with regexps. What do you think about us instead changing the `User-Agent` header to identify specifically this is Terraform (rather than just the Go HTTP client) and including the version number in that string? Not sure what use-case you have in mind, but if this works I think it's a safer change to make and something that is hopefully generally useful for those using access logs for usage stats.

**Comment by antonbabenko** [comment]
Yes, setting `User-Agent` with Terraform version (in form of string) is very good alternative also.
